docs/how-to/partitions.rst

```diff
@@ -0,0 +1,123 @@
+
+Partition names and namespace names must consist of `only` lower-case alphabetic characters, unless a partition exists under a namespace, in which case it may also contain hyphen characters, though the first and last characters must still be alphabetic.
+
+.. _app_changes:
+
+Required application changes
+============================
+
+To add partition support to an application, two basic changes are needed to the ``craft-application``.
+
+#. Enable the feature
+
+   When creating the :class:`AppMetadata <craft_application.AppMetadata>`, specify that the application will use the partitions feature::
+
+     from craft_application import AppMetadata, AppFeatures
+
+     APP_METADATA = AppMetadata(
+       ...
+       features=AppFeatures(enable_partitions=True),
+     )
+
+#. Define the list of partitions
+
+   We need to tell the :class:`LifecycleManager <craft_parts.LifecycleManager>` class about our partitions, but applications do not usually directly instantiate the LifecycleManager.
+
+   Inside your :class:`Application <craft_application.Application>` class, the :class:`ServiceFactory <craft_application.ServiceFactory>` can be told to instantiate the :class:`LifecycleService <craft_application.LifecycleService` with specific ``custom_args`` by calling ``set_kwargs``.
+
+   Override the ``_configure_services`` method, and use ``set_kwargs`` to define the partitions that will eventually be passed to the :class:`LifecycleManager <craft_parts.LifecycleManager>`::
+
+     @override
+     def _configure_services(self, provider_name: str | None) -> None:
+       super()._configure_services(provider_name)
+
+       ...
+
+       self.services.set_kwargs(
+         "lifecycle",
+         ...
+         partitions=["default", "kernel", "component/bar-baz"],
+       )
+
+Using the partitions
+====================
+
+Partitions cannot be used until `after you have configured your application <#app-changes>`_.
+
+In configuration yaml lifecycle
+-------------------------------
+
+Defined partitions may be referenced in the ``organize``, ``stage``, and ``prime`` sections of your configuration yaml files::
+
+  organize:
+    <source-path>: (<partition>)/<path>
+  stage:
+    - (<partition>)/<path>
+  prime:
+    - (<partition>)/<path>
+
+Paths in the configuration yaml not beginning with a partition label will implicitly use the default partition.
+
+The source path of an ``organize`` entry can only be from the default partition.  For example::
+
+  organize:
+    (kernel)/usr/local/bin/hello: bin/hello
+
+  Cannot organize files from 'kernel' partition.
+  Files can only be organized from the 'default' partition
+
+When the ``stage`` and ``prime`` keywords are not provided for a part, craft-parts' default behavior is to stage and prime all files for the part in all partitions.
+
+(If a stage or prime filter `is` applied to a partition, the default behavior will not be affected for the other partitions.)
```

Comment:
needs double ticks:

```suggestion
(If a stage or prime filter ``is`` applied to a partition, the default behavior will not be affected for the other partitions.)
```

Will you be so kind as to sneak this into your PR?

```diff
diff --git a/.github/workflows/tests.yaml b/.github/workflows/tests.yaml
index 1bff626..31c3fa2 100644
--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -42,6 +42,9 @@ jobs:
       - name: Run pydocstyle
         run: |
           make test-pydocstyle
+      - name: Run lint-docs
+        run: |
+          tox run -e lint-docs
       - name: Run pyright
         run: |
           sudo snap install --classic node
```

craft-parts hasn't been rebased onto starbase so this is a bandaid to ensure we don't start snowballing documentation problems from a lack of linting.

It looks like there are 3 small errors in existing documentation that should be very quick to fix.

Ideally you would break all lines at 80 characters, but a ton of docs here are not conforming to that so we can deal with that later.

Reply:
Thanks @mr-cal!  In your suggestion, the single backticks were intentional - I want that "is" to be italicized, not monospaced.

And yep no problem, I'll add the doc linting band-aid and fix the other linting issues.

Reply:
I think `*` is the right way to do italics or "emphasis" for RST

Reply:
Thanks, fixed